System/RTEError.cpp

```diff
@@ -7,73 +7,403 @@
+
+			// This bullshit makes the error string and returns the size because we can't know it in advance. Don't think we actually care about the size when we construct string from a buffer but whatever.
+			size_t messageSize = FormatMessage(messageFlags, nullptr, errorMessageID, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), reinterpret_cast<LPSTR>(&messageBuffer), 0, nullptr);
+			std::string message(messageBuffer, messageSize);
+			LocalFree(messageBuffer);
+			return message;
+		};
+
+		// Returns a string with the type of the exception from the passed in code.
+		static auto getExceptionDescriptionFromCode = [](const DWORD &exceptCode) -> std::string {
+			switch (exceptCode) {
+				case EXCEPTION_ACCESS_VIOLATION:			return "EXCEPTION_ACCESS_VIOLATION";
+				case EXCEPTION_ARRAY_BOUNDS_EXCEEDED:		return "EXCEPTION_ARRAY_BOUNDS_EXCEEDED";
+				case EXCEPTION_BREAKPOINT:					return "EXCEPTION_BREAKPOINT";
+				case EXCEPTION_DATATYPE_MISALIGNMENT:		return "EXCEPTION_DATATYPE_MISALIGNMENT";
+				case EXCEPTION_FLT_DENORMAL_OPERAND:		return "EXCEPTION_FLT_DENORMAL_OPERAND";
+				case EXCEPTION_FLT_DIVIDE_BY_ZERO:			return "EXCEPTION_FLT_DIVIDE_BY_ZERO";
+				case EXCEPTION_FLT_INEXACT_RESULT:			return "EXCEPTION_FLT_INEXACT_RESULT";
+				case EXCEPTION_FLT_INVALID_OPERATION:		return "EXCEPTION_FLT_INVALID_OPERATION";
+				case EXCEPTION_FLT_OVERFLOW:				return "EXCEPTION_FLT_OVERFLOW";
+				case EXCEPTION_FLT_STACK_CHECK:				return "EXCEPTION_FLT_STACK_CHECK";
+				case EXCEPTION_FLT_UNDERFLOW:				return "EXCEPTION_FLT_UNDERFLOW";
+				case EXCEPTION_ILLEGAL_INSTRUCTION:			return "EXCEPTION_ILLEGAL_INSTRUCTION";
+				case EXCEPTION_IN_PAGE_ERROR:				return "EXCEPTION_IN_PAGE_ERROR";
+				case EXCEPTION_INT_DIVIDE_BY_ZERO:			return "EXCEPTION_INT_DIVIDE_BY_ZERO";
+				case EXCEPTION_INT_OVERFLOW:				return "EXCEPTION_INT_OVERFLOW";
+				case EXCEPTION_INVALID_DISPOSITION:			return "EXCEPTION_INVALID_DISPOSITION";
+				case EXCEPTION_NONCONTINUABLE_EXCEPTION:	return "EXCEPTION_NONCONTINUABLE_EXCEPTION";
+				case EXCEPTION_PRIV_INSTRUCTION:			return "EXCEPTION_PRIV_INSTRUCTION";
+				case EXCEPTION_SINGLE_STEP:					return "EXCEPTION_SINGLE_STEP";
+				case EXCEPTION_STACK_OVERFLOW:				return "EXCEPTION_STACK_OVERFLOW";
+				default:
+					return "UNKNOWN EXCEPTION";
+			}
+		};
+
+		// Attempts to get a symbol name from the exception address.
+		static auto getSymbolNameFromAddress = [](HANDLE &procHandle, const size_t &exceptAddr) {
+			if (SymInitialize(procHandle, nullptr, TRUE)) {
+				SymSetOptions(SYMOPT_UNDNAME | SYMOPT_DEFERRED_LOADS);
+
+				char symbolBuffer[sizeof(SYMBOL_INFO) + MAX_SYM_NAME * sizeof(char)];
+				PSYMBOL_INFO symbolInfo = reinterpret_cast<PSYMBOL_INFO>(symbolBuffer);
+
+				symbolInfo->SizeOfStruct = sizeof(SYMBOL_INFO);
+				symbolInfo->MaxNameLen = MAX_SYM_NAME;
+
+				if (SymFromAddr(procHandle, exceptAddr, nullptr, symbolInfo)) {
```

Comment:
Am confused, does this just always fail?
You have an if check that says "this is empty for some reason" else you're unable to get the symbol name at the address. 

But there's no earlier return?

Reply:
I am confused of your confusion.
This returns bool on success.
It might succeed and still end up without a symbol name, for reasons unknown to man.

Does it always fail for you or something?

Reply:
Oh goddamnit, I'm the dumb and didn't read the latter part of the empty ternary. Pls ignore.